packages/jmix-react-antd/src/ui/entity-filter/EntityFilter.tsx

```diff
@@ -0,0 +1,135 @@
+import React, {useState} from "react";
+import {createAntdFormValidationMessages, Msg} from "@haulmont/jmix-react-web";
+import {Button, Form, Input, Select, Space} from "antd";
+import {useForm} from "antd/es/form/Form";
+import {useIntl} from "react-intl";
+import {CloseOutlined, PlusOutlined} from "@ant-design/icons";
+import {
+  FilterChangeCallback,
+  getPropertyInfo,
+  JmixEntityFilter, MetaClassInfo,
+  useMetadata
+} from "@haulmont/jmix-react-core";
+
+type EntityFilterProps = {
+  entityName: string,
+  propertyNames: string[],
+  onFilterApply?: FilterChangeCallback,
+  className?: string,
+}
+
+export const EntityFilter = ({entityName, propertyNames, onFilterApply, className}: EntityFilterProps) => {
+  const intl = useIntl();
+  const [form] = useForm();
+  const metadata = useMetadata();
+  const [selectedProperty, setSelectedProperty] = useState<string>();
+  const [availableProperties, setAvailableProperties] = useState<string[]>(getAvailableProperties(metadata.entities, entityName, propertyNames));
+
+  const availablePropertiesOptions = availableProperties.map(propertyName => ({
+    label: <Msg entityName={entityName} propertyName={propertyName}/>,
+    value: propertyName,
+  }));
+
+  const onAddProperty = (addFilter: (defaultValue: JmixEntityFilter) => void) => {
+    addFilter({[selectedProperty!]: {_eq: ""}});
```

Comment:
We are reusing a helper from `DataTable`. It is better to move `addFilter` to common utils (`src/util`).

Reply:
`onAddProperty` receives `addFilter` callback for adding filters, but managing filters state in `EntityFilter` is done by [Form.List](https://ant.design/components/form/#Form.List). `addFilter` helper from `DataTable` isn't used here